packages/core/src/hooks/useEmbrace.ts

```diff
@@ -0,0 +1,43 @@
+import {useEffect, useState} from "react";
+
+import {SDKConfig} from "../interfaces/common";
+
+import {initialize} from "./../index";
+
+const useEmbrace = (sdkConfig: SDKConfig, debug: boolean = true) => {
+  // States
+  const [isPending, setIsPending] = useState(true);
+  const [isStarted, setIsStarted] = useState(false);
+
+  /**
+   * Init SDK
+   */
+  useEffect(() => {
+    const initSDK = async () => {
+      const config = {
+        sdkConfig,
+        debug,
+      };
+
+      try {
+        const isEmbraceStarted = await initialize(config);
+
+        setIsPending(false);
+        setIsStarted(isEmbraceStarted);
+      } catch (error) {
+        setIsPending(false);
+      }
```

Comment:
could use `finally` here instead since it seems we'll always want pending turned off:

```suggestion
      try {
        const isEmbraceStarted = await initialize(config);
        setIsStarted(isEmbraceStarted);
      } finally {
        setIsPending(false);
      }
```